Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -67,43 +79,76 @@ public final class CoreLocationMiddleware: Middleware {
+            } else {
+                return
+            }
+        case .requestAuthorizationType:
+            switch getState?().authzType {
+            case .always:
+                manager.requestAlwaysAuthorization()
+            case .whenInUse:
+                manager.requestWhenInUseAuthorization()
+            case .none:
+                break
+            }
+        case .requestPosition: manager.requestLocation()
+        case .requestDeviceCapabilities: delegate.output?.dispatch(getDeviceCapabilities())
         default: return
         }
     }
+    
     func startMonitoring() {
-        switch getState?() {
-        case .authorized?:
-            manager.startUpdatingLocation()
-        default:
-            // requestAlwaysAuthorization or requestWhenInUseAuthorization could be decided on
-            // the middleware init or as payload for startMonitoring
-            manager.requestAlwaysAuthorization()
+        let stateType = getState?().authzType
+        let stateStatus = getState?().authzStatus
+        
+        switch stateType {
+        case .always:
+            if CLAuthorizationStatus.authorizedAlways != stateStatus {
+                manager.requestAlwaysAuthorization()
+            }
+        case .whenInUse:
+            if ![CLAuthorizationStatus.authorizedAlways, CLAuthorizationStatus.authorizedWhenInUse].contains(stateStatus) {
+                manager.requestWhenInUseAuthorization()
+            }
+        case .none:
+            break
         }
+        manager.startUpdatingLocation()
     }
+    
     func stopMonitoring() {
         manager.stopUpdatingLocation()
     }
+    
+    func getDeviceCapabilities() -> LocationAction {
+        .gotDeviceCapabilities(
+            DeviceCapabilities(
+                isSignificantLocationChangeAvailable: CLLocationManager.significantLocationChangeMonitoringAvailable(),
+                isHeadingAvailable: CLLocationManager.headingAvailable(),
+                isRegionMonitoringAvailable: CLLocationManager.isMonitoringAvailable(for: CLCircularRegion.self),
+                isRangingAvailable: CLLocationManager.isRangingAvailable(),
+                isLocationServiceAvailable: CLLocationManager.locationServicesEnabled()
+            )
+        )
+    }
 }
 
- class CLDelegate: NSObject, CLLocationManagerDelegate {
+class CLDelegate: NSObject, CLLocationManagerDelegate {
     
     var output: AnyActionHandler<LocationAction>? = nil
     
     func locationManager(_ manager: CLLocationManager, didChangeAuthorization status: CLAuthorizationStatus) {
-        switch status {
-        case .notDetermined:
-            output?.dispatch(.authorizationUnknown)
-        case .denied, .restricted:
-            output?.dispatch(.unauthorized)
-        case .authorizedAlways, .authorizedWhenInUse:
-            output?.dispatch(.authorized)
-        @unknown default:
+        if #available(iOS 14.0, *) {
+            output?.dispatch(getAuthzStatus(status: status))
```

Comment:
Why we only dispatch this for iOS 14? Probably on iOS 13 our LocationState will only show the wrong authorization status forever, right?

Reply:
Not needed. Leftover from a previous version where I had the call to `authorizationStatus` inside the delegate method. I submitted a new commit.